### PR TITLE
feat: Add delete override implementation

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
### Description of your changes

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/utils/validator/resourceoverride.go [hunk 1]
```diff
@@ -102,9 +102,16 @@ func validateOverridePolicy(policy *fleetv1alpha1.OverridePolicy) error {
 				}
 			}
 		}
+		switch rule.OverrideType {
```
Comment (zhiying-lin) [inline pkg/utils/validator/resourceoverride.go:105]: can be replaced by using cel

for example,

// +kubebuilder:validation:XValidation:rule="self.overrideType == 'JSONPatch' || self.jsonPatchOverrides.size() == 0 ",message="jsonPatchOverrides must be empty when type is not JSONPatch"

File: pkg/controllers/workgenerator/override.go [hunk 2]
```diff
@@ -131,13 +137,12 @@ func (r *Reconciler) applyOverrides(resource *placementv1beta1.ResourceContent,
...
 			klog.ErrorS(err, "Failed to apply the override rules", "clusterResourceOverrideSnapshot", klog.KObj(snapshot))
-			return err
+			return false, err
 		}
 	}
 	klog.V(2).InfoS("Applied clusterResourceOverrideSnapshots", "resource", klog.KObj(&uResource), "numberOfOverrides", len(croMap[key]))
 
-	// If the resource is selected by both ClusterResourceOverride and ResourceOverride, ResourceOverride will win when
-	// resolving conflicts.
+	// If the resource is selected by both ClusterResourceOverride and ResourceOverride, ResourceOverride will win when resolving conflicts.
```
Comment (zhiying-lin) [inline pkg/controllers/workgenerator/override.go:145]: what happens for a namespace scope resource which is selected by both cro and ro?

for example, cro deletes the resource, and ro overrides the resource.

File: pkg/controllers/workgenerator/override_test.go [hunk 3]
```diff
@@ -730,11 +808,11 @@ func TestApplyOverrides_namespacedScopeResource(t *testing.T) {
 				Kind:    "Deployment",
 			}: true,
 		},
-		IsClusterScopedResource: false,
+		IsClusterScopedResource: true,
```
Comment (zhiying-lin) [inline pkg/controllers/workgenerator/override_test.go:811]: should be "false"? deployment is not a cluster scoped resource.

File: pkg/controllers/workgenerator/override_test.go [hunk 4]
```diff
@@ -1253,21 +1410,73 @@ func TestApplyOverrides_namespacedScopeResource(t *testing.T) {
 			},
 			wantErr: controller.ErrUserError,
 		},
+		{
```
Comment (zhiying-lin) [inline pkg/controllers/workgenerator/override_test.go:1413]: add more tests about delete type:
- cro delete, ro patch
- cro patch, ro delete
- cro delete, ro delete

File: pkg/controllers/workgenerator/controller.go [hunk 5]
```diff
@@ -447,7 +452,7 @@ func (r *Reconciler) syncAllWork(ctx context.Context, resourceBinding *fleetv1be
 				activeWork[work.Name] = work
 				newWork = append(newWork, work)
 			} else {
-				simpleManifests = append(simpleManifests, fleetv1beta1.Manifest(selectedResource))
+				simpleManifests = append(simpleManifests, fleetv1beta1.Manifest(*selectedResource))
 			}
 		}
 		if len(simpleManifests) == 0 {
```
Comment (zhiying-lin) [inline pkg/controllers/workgenerator/controller.go:458]: this log needs to be updated.

File: pkg/utils/validator/resourceoverride.go (hunk 1 above)
Comment (ryanzhang-oss) [reply pkg/utils/validator/resourceoverride.go:105]: good to know but unless we remove this validator, we need to fix it even with the CEL, right?

Comment (zhiying-lin) [reply pkg/utils/validator/resourceoverride.go:105]: yeah, but the fix is much simpler, just need to add the type check.

File: pkg/controllers/workgenerator/override_test.go [hunk 6]
```diff
@@ -658,13 +937,65 @@ func TestApplyOverrides_clusterScopedResource(t *testing.T) {
...
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: controller.ErrUserError,
+		},
+		{
+			name: "delete during the clusterResourceOverride",
```
[the remark this answers is not in the document]
Comment (ryanzhang-oss) [reply pkg/controllers/workgenerator/override_test.go:950]: isn't test 1719  the delete after patch. There is another delete after patch case for a field earlier.

File: pkg/utils/validator/resourceoverride.go (hunk 1 above)
Comment (ryanzhang-oss) [reply pkg/utils/validator/resourceoverride.go:105]: I think there is some advantage to keep the logic (and test) together intead of spliting them into two places.

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**